microdnf-plugins/changelog_plugin/changelog.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "changelog.hpp"
+
+#include "microdnf/context.hpp"
+
+#include <libdnf/conf/option_string.hpp>
+#include <libdnf/rpm/package.hpp>
+#include <libdnf/rpm/package_query.hpp>
+#include <libdnf/rpm/package_set.hpp>
+
+#include <ctime>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+
+namespace microdnf {
+
+
+using namespace libdnf::cli;
+
+
+ChangelogCommand::ChangelogCommand(Command & parent) : Command(parent, "changelog") {
+    auto & ctx = static_cast<Context &>(get_session());
+    auto & parser = ctx.get_argument_parser();
+
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_short_description("Show package changelogs");
+
+    since_option = dynamic_cast<libdnf::OptionNumber<std::int64_t> *>(
+        parser.add_init_value(std::unique_ptr<libdnf::OptionNumber<std::int64_t>>(
+            new libdnf::OptionNumber<int64_t>(0, [](const std::string & value) {
+                struct tm time_m;
+                std::istringstream ss(value);
+                ss >> std::get_time(&time_m, "%Y-%m-%d");
+                if (ss.fail()) {
+                    throw std::runtime_error(fmt::format("Invalid date: {}", value));
+                }
+                return static_cast<int64_t>(mktime(&time_m));
+            }))));
+
+    count_option = dynamic_cast<libdnf::OptionNumber<std::int32_t> *>(
+        parser.add_init_value(std::unique_ptr<libdnf::OptionNumber<std::int32_t>>(new libdnf::OptionNumber<int>(0))));
+
+    upgrades_option = dynamic_cast<libdnf::OptionBool *>(
+        parser.add_init_value(std::unique_ptr<libdnf::OptionBool>(new libdnf::OptionBool(false))));
+
+    auto since = parser.add_new_named_arg("since");
+    since->set_long_name("since");
+    since->set_short_description(
+        "Show changelog entries since DATE. To avoid ambiguosity, YYYY-MM-DD format is required.");
```

Comment:
I was suggesting to make this one shorter:
> Show changelog entries since YYYY-MM-DD

Though if you don't like how short it is, something like this is also fine:
> Show changelog entries since date in the YYYY-MM-DD format

The note about ambiguity seems redundant to me, and we shouldn't say a particular format is required, we should just specify the format. At the very least, `s/ambiguosity/ambiguity/` :slightly_smiling_face: